core/src/main/java/com/sequenceiq/cloudbreak/converter/stack/cluster/ServiceEndpointCollector.java

```diff
@@ -0,0 +1,189 @@
+        List<Port> ports = NetworkUtils.getAllPorts();
+        try {
+            JsonNode hostGroupsNode = blueprintValidator.getHostGroupNode(blueprint);
+            Map<String, HostGroup> hostGroupMap = blueprintValidator.createHostGroupMap(hostGroups);
+            for (JsonNode hostGroupNode : hostGroupsNode) {
+                String hostGroupName = blueprintValidator.getHostGroupName(hostGroupNode);
+                JsonNode componentsNode = blueprintValidator.getComponentsNode(hostGroupNode);
+                HostGroup actualHostgroup = hostGroupMap.get(hostGroupName);
+                final String serviceAddress;
+                if (actualHostgroup.getConstraint().getInstanceGroup() != null) {
+                    InstanceMetaData next = actualHostgroup.getConstraint().getInstanceGroup().getNotDeletedInstanceMetaDataSet().iterator().next();
+                    serviceAddress = next.getPublicIpWrapper();
+                } else {
+                    serviceAddress = actualHostgroup.getHostMetadata().iterator().next().getHostName();
+                }
+                for (JsonNode componentNode : componentsNode) {
+                    String componentName = componentNode.get("name").asText();
+                    StackServiceComponentDescriptor componentDescriptor = stackServiceComponentDescs.get(componentName);
+                    Map<String, String> collectedPorts = collectServicePorts(ports, ambariIp, serviceAddress, componentDescriptor, cluster);
+                    result.putAll(collectedPorts);
+                }
+            }
+        } catch (Exception ignored) {
+            return result;
+        }
+        return result;
+    }
+
+    public String getAmbariServerUrl(Cluster cluster, String ambariIp) {
+        String orchestrator = cluster.getStack().getOrchestrator().getType();
+        if (ambariIp != null) {
+            Gateway gateway = cluster.getGateway();
+            if (YARN.equals(orchestrator)) {
+                return String.format("http://%s:8080", ambariIp);
+            } else {
+                if (gateway != null) {
+                    Optional<GatewayTopology> gatewayTopologyWithAmbari = getGatewayTopologyWithAmbari(gateway);
+                    return gatewayTopologyWithAmbari
+                            .map(gt -> getAmbariUrlFromGatewayTopology(ambariIp, gateway, gt))
+                            .orElse(String.format("https://%s/ambari/", ambariIp));
+                    // TODO: ask if its good when ambari is not exposed in any topology in the gateway
+                } else {
+                    return String.format("https://%s/ambari/", ambariIp);
+                }
+            }
+        }
+        return null;
+    }
+
+    private Map<String, String> collectServicePorts(Iterable<Port> ports, String ambariIp, String serviceAddress,
+            StackServiceComponentDescriptor componentDescriptor, Cluster cluster) {
+        Map<String, String> collectedPorts = new HashMap<>();
```

Comment:
Unnecessary new instance of map here, some cases we return an empty map which is equals with Collections.emptyMap() which doesn't require new instance